File: Search_insert_position/Search_insert_position.py
def search_insert_position(nums, target):
    result = 0

    for num in range(len(nums)):
        if target == nums[num]:
            result = num
        elif target != num:
            if target > nums[len(nums)-1]:
                result = len(nums)
            elif target < nums[0]:
                result = 0
            elif target > nums[num-1] and target < nums[num]:
                    result = num

    print(result)

search_insert_position([1,3,5,6], 5)
search_insert_position([1,3,5,6], 2)
search_insert_position([1,3,5,6],7)
search_insert_position([1], 2) 

File: Search_insert_position/searchInsertPosition.java
public class searchInsertPosition {
    public static void main(String[] args) {
        searchInsertPosition(new int[] {1, 3, 5, 6}, 5); // Expected output: 2
        searchInsertPosition(new int[] {1, 3, 5, 6}, 2); // Expected output: 1
        searchInsertPosition(new int[] {1, 3, 5, 6}, 7); // Expected output: 4
        searchInsertPosition(new int[] {1}, 2);          // Expected output: 1
    }

    public static void searchInsertPosition(int[] nums, int target) {
        int result = nums.length; // Default to insert at the end

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == target) {
                result = i;
                break;
            } else if (nums[i] > target) {
                result = i;
                break;
            }
        }

        System.out.println(result);
    }
}